pkg/imgpkg/bundle/bundle.go

```diff
@@ -19,48 +21,116 @@ const (
 )
 
 type Bundle struct {
-	plainImg     *plainimg.PlainImage
-	imgRetriever ctlimg.ImagesMetadata
+	plainImg      *plainimg.PlainImage
+	imgRetriever  ctlimg.ImagesMetadata
+	numSubBundles int
 }
 
 func NewBundle(ref string, imagesMetadata ctlimg.ImagesMetadata) *Bundle {
-	return &Bundle{plainimg.NewPlainImage(ref, imagesMetadata), imagesMetadata}
+	return &Bundle{plainimg.NewPlainImage(ref, imagesMetadata), imagesMetadata, 0}
 }
 
 func NewBundleFromPlainImage(plainImg *plainimg.PlainImage, imagesMetadata ctlimg.ImagesMetadata) *Bundle {
-	return &Bundle{plainImg, imagesMetadata}
+	return &Bundle{plainImg, imagesMetadata, 0}
 }
 
 func (o *Bundle) DigestRef() string { return o.plainImg.DigestRef() }
 func (o *Bundle) Repo() string      { return o.plainImg.Repo() }
 func (o *Bundle) Tag() string       { return o.plainImg.Tag() }
 
-func (o *Bundle) Pull(outputPath string, ui ui.UI) error {
+func (o *Bundle) Pull(outputPath string, ui goui.UI, recursive bool) error {
+	return o.pull(recursive, outputPath, "", map[string]bool{}, ui)
+}
+
+func (o *Bundle) pull(recursive bool, baseOutputPath string, bundlePath string, imagesProcessed map[string]bool, ui goui.UI) error {
 	img, err := o.checkedImage()
 	if err != nil {
 		return err
 	}
 
-	ui.BeginLinef("Pulling bundle '%s'\n", o.DigestRef())
+	if o.rootBundle(bundlePath) {
+		ui.BeginLinef("Pulling bundle '%s'\n", o.DigestRef())
+	} else {
+		ui.BeginLinef("Pulling Nested bundle '%s'\n", o.DigestRef())
```

Comment:
nested shouldnt be capitlized.

Reply:
attention. to. detail. i'll update this.